project-base/storefront/components/PaymentsInOrderSelect/PaymentsInOrderSelect.tsx

```diff
@@ -0,0 +1,125 @@
+import { PaymentsInOrderSelectItem } from './PaymentsInOrderSelectItem';
+import { InfoIcon, SpinnerIcon } from 'components/Basic/Icon/IconsSvg';
+import { Button } from 'components/Forms/Button/Button';
+import { GoPayGateway } from 'components/Pages/Order/PaymentConfirmation/Gateways/GoPayGateway';
+import { useChangePaymentInOrder } from 'components/PaymentsInOrderSelect/helpers';
+import { SimplePaymentFragmentApi, useOrderAvailablePaymentsQueryApi } from 'graphql/generated';
+import { useIsPaymentWithCartAvailable } from 'hooks/cart/useIsPaymentWithCartAvailable';
+import useTranslation from 'next-translate/useTranslation';
+import { useEffect, useState } from 'react';
+import { PaymentTypeEnum } from 'types/payment';
+
+type PaymentsInOrderSelectProps = {
+    orderUuid: string;
+    withRedirectAfterChanging?: boolean;
+    paymentTransactionCount: number;
+};
+
+export const PaymentsInOrderSelect: FC<PaymentsInOrderSelectProps> = ({
+    orderUuid,
+    withRedirectAfterChanging,
+    paymentTransactionCount,
+}) => {
+    const { t } = useTranslation();
+
+    const { isChangePaymentInOrderFetching, changePaymentInOrderHandler } = useChangePaymentInOrder();
+    const [selectedPaymentSwiftForChange, setSelectedPaymentSwiftForChange] = useState<string | undefined | null>();
+    const [selectedPaymentForChange, setSelectedPaymentForChange] = useState<SimplePaymentFragmentApi>();
+    const [isGoPayVisible, setIsGoPayVisible] = useState(false);
+
+    const [{ data: orderAvailablePaymentsData, fetching: isOrderAvailablePaymentsFetching }] =
+        useOrderAvailablePaymentsQueryApi({
+            variables: { orderUuid },
+        });
+
+    const isPaymentWithCardAvailable = useIsPaymentWithCartAvailable(paymentTransactionCount);
+    const currentOrderPayment = orderAvailablePaymentsData?.orderPayments.currentPayment;
+    const isSelectedPaymentEqualsOrderPayment = selectedPaymentForChange?.uuid === currentOrderPayment?.uuid;
```

Comment:
Either `isSelectedPaymentEqualToOrderPayment` or `doesSelectedPaymentEqualOrderPayment`, to make it grammatically correct